_modules/conditionals.md

```diff
@@ -6,3 +6,119 @@ short_title: Buttons and Switches
+
+```c
+if (buttons_get(1)) {
+    leds_set_color(1, 255, 0, 0);
+}
+```
+
+This block of code tells our board to turn on LED1 _if_ Button1 is pressed. The function `buttons_get()` can be used as a **conditional** because it returns a `true` or `false` value that indicates whether the button that was passed into it (in this case Button1) is pressed. Notice that in an **if statement** the **conditional** is placed in parenthesis and the commands to execute if it is true are placed in curly braces.
+
+
+## Examples
+If you copy the above code into your program, and then try to upload it to your board and press the button, LED1 won't turn on! The reason why is that the _if_ statement is run immediately when you upload your code and it is only run once. In order to get the board to _continuously check_ the state of Button1 and turn LED1 on if it is pressed, we need to wrap our `if` statement in a `while` loop:
+
+```c
+while(true) {
+    if (buttons_get(1)) {
+        leds_set_color(1, 255, 0, 0);
+    } else {
+        leds_set_color(1, 0, 0, 0);
+    }
+}
+```
+
+Notice that we've added an `else` block to our `if` statement. This makes it so that if we let go of Button1, LED1 will turn off again.
+
+Combining this with the skills you've learned with variables and `for` or `while` loops begins to introduce all kinds of fun possibilities for you to explore. For example, we could get the board to only check the state of our button for the first 10 seconds after we program it, and then light up LED15 after those 10 seconds, like this:
+
+```c
+for(int count = 0; count < 100; count++) {
+    if (buttons_get(1)) {
+        leds_set_color(1, 255, 0, 0);
+    } else {
+        leds_set_color(1, 255, 0, 0);
+    }
+    delay(100);
+}
+leds_set_color(15, 255, 0, 0);
+```
+
+This code block only checks if Button1 is pressed in 0.1 second increments for 10 seconds. 0.1 seconds is faster than the average person can press the button, so it will work just fine, and once the `for` loop is done, you can continue writing your code as normal.
+
+Now let's talk about the `else if` block. If we want to turn LED1 red when only one button is pressed, yellow when two buttons are pressed, or green when all three buttons are pressed, we could use the following:
+
+```c
+while (true) {
+    if (buttons_get(1) && buttons_get(2) && buttons_get(3)) {
+        leds_set_color(1, 0, 255, 0);
+    } else if (buttons_get(1) && buttons_get(2)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(2) && buttons_get(3)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(1) && buttons_get(3)) {
+        leds_set_color(1, 255, 255, 0);
+    } else if (buttons_get(1)) {
+        leds_set_color(1, 255, 0, 0);
+    } else if (buttons_get(2)) {
+        leds_set_color(1, 255, 0, 0);
+    } else if (buttons_get(3)) {
+        leds_set_color(1, 255, 0, 0);
+    } else {
+        leds_set_color(1, 0, 0, 0);
+    }
+}
+```
```

Comment:
Can we make this example a bit simpler?  I worry it will take too long to go through and we will lose their attention a bit.

Reply:
Done